eng/WpfArcadeSdk/tools/Pbt.targets

```diff
@@ -1,104 +1,113 @@
+      (c) The assembly is placed at $(_PresentationBuildTasksAssembly)
+      (d) This is all done prior to the first ever use of a task from within this assembly by Microsoft.WinFx.targets
+  -->
+  <Target Name="EnsurePbtPath"
+          Condition="'$(InternalMarkupCompilation)'=='true' And Exists('$(LocalMicrosoftWinFXTargets)')"
+          DependsOnTargets="$(EnsurePbtPathDependsOn)"
+          BeforeTargets="ResolveProjectReferences">
+    <MakeDir Condition="!Exists('$(PbtDir)')"
+             Directories="$(PbtDir)" />
+    
+    <MSBuild Projects="$(WpfSourceDir)PresentationBuildTasks\PresentationBuildTasks.csproj"
+             Condition="!Exists('$(_PresentationBuildTasksAssembly)') And Exists('$(LocalMicrosoftWinFXTargets)')"
+             Properties="CopyTransitiveReferences=true;PublishDir=$(PbtDir);TargetFramework=$(PbtTfm);Platform=AnyCPU"
+             Targets="Clean;Build;Publish" />
+  </Target>
+
+  <Target Name="PreventResourcesSourceGenerationForPbtGeneratedResource" 
+          AfterTargets="MainResourcesGeneration"
+          Condition="'$(InternalMarkupCompilation)'=='true'">
+    <!-- 
+      Add the following metadata to $(IntermediateOutputPath)$(_ResourceNameInMainAssembly) within @(EmbeddedResource)
+              <XlfInput>false</XlfInput>
+              <GenerateSource>false</GenerateSource
+              
+      This ensures that PBT generated resources are not inadvertently picked up by Xlf source generator.
+    -->
+    <ItemGroup Condition="'$(_ResourceNameInMainAssembly)'!=''">
+      <EmbeddedResource Condition="'%(EmbeddedResource.Identity)'=='$(IntermediateOutputPath)$(_ResourceNameInMainAssembly)'">
+        <XlfInput>false</XlfInput>
+        <GenerateSource>false</GenerateSource>
+      </EmbeddedResource>
+    </ItemGroup>
+  </Target>
+
+  <!-- 
+    If local PresentationBuildTasks project is present, then import WinFX.targets from local sources;
+    otherwise import Microsoft.NET.Sdk.WindowsDesktop.targets from Microsoft.NET.Sdk.WindowsDesktop
+  -->
+  <Import Project="$(WpfSourceDir)PresentationBuildTasks\Microsoft.WinFX.targets" 
+          Condition="'$(InternalMarkupCompilation)'=='true' And Exists('$(LocalMicrosoftWinFXTargets)') "/>
+
+  <!-- 
+    _WindowsDesktopSdkTargetFrameworkVersionFloor is defined in Microsoft.NET.WindowDesktop.props. 
+    This needs to be defined before Microsoft.NET.Sdk.WindowsDesktop.targets is imported. 
+  -->
+  <PropertyGroup Condition="'$(InternalMarkupCompilation)'=='true' And !Exists('$(LocalMicrosoftWinFXTargets)') ">
+    <_WindowsDesktopSdkTargetFrameworkVersionFloor Condition="'$(_WindowsDesktopSdkTargetFrameworkVersionFloor)' == ''">3.0</_WindowsDesktopSdkTargetFrameworkVersionFloor>
+  </PropertyGroup>
+  <Import Sdk="Microsoft.NET.Sdk.WindowsDesktop"
+          Project="../targets/Microsoft.NET.Sdk.WindowsDesktop.targets"
+          Condition="'$(InternalMarkupCompilation)'=='true' And !Exists('$(LocalMicrosoftWinFXTargets)') "/>
+

```

Comment:
Can you add explanations for salient changes? /cc @ryalanms would you mind reviewing this change?

Reply:
Why aren't the diffs working correctly?

Reply:
yeah, when building the theme dlls we get errors about `WindowsBase` already being in the reference list. i don't know exactly why this wasn't happening internally, but i looked at the binlog and saw that a reference to `WindowsBase` was coming from `Microsoft.NetCore.App.Ref`. this would then manifest itself in an exception being thrown by PBT about `WindowsBase` already being loaded in the `MetadataLoadContext`. So maybe when building internally we weren't picking up the `WindowsBase` that came from GitHub? Regardless, i feel like we never want the `WindowsBase` reference from `Microsoft.NetCore.App.Ref`

Reply:
Could it be that the project in question is not using `NetCoreReference` carefully, and thus it is inheriting `WindowsBase `from `Microsoft.NetCore.App.Ref `when it shouldn't?

Reply:
i'll add a comment

Reply:
When a project uses `NetCoreReference`, only those references that are reqeusted intentionally are used, which means that WindowsBase is not included. It's possible that the project where you had problems forgot to use `NetCoreReference` to enumerate the references it needs, and instead inherits _all_ references from `Microsoft.NetCore.App`. If this is what's happening, then the preferred solution is to fixup the project - identify the smallest set of `NetCoreReference` items and incorporate it into the csproj directly.

Reply:
It looks like the project already does that?

```
 <!-- Minimal set of .NET core references -->
  <ItemGroup>
    <NetCoreReference Include="netstandard" />
    <NetCoreReference Include="System.Collections" />
    <NetCoreReference Include="System.Collections.Generic" />
    <NetCoreReference Include="System.Collections.NonGeneric" />
    <NetCoreReference Include="System.ComponentModel" />
    <NetCoreReference Include="System.ComponentModel.Primitives" />
    <NetCoreReference Include="System.Diagnostics.Debug" />
    <NetCoreReference Include="System.Diagnostics.Process" />
    <NetCoreReference Include="System.Diagnostics.Tools" />
    <NetCoreReference Include="System.Diagnostics.TraceSource" />
    <NetCoreReference Include="System.Drawing.Common" />
    <NetCoreReference Include="System.Drawing.Primitives" />
    <NetCoreReference Include="System.IO" />
    <NetCoreReference Include="System.IO.FileSystem" />
    <NetCoreReference Include="System.ObjectModel" />
    <NetCoreReference Include="System.Resources.ResourceManager" />
    <NetCoreReference Include="System.Runtime.Extensions" />
    <NetCoreReference Include="System.Runtime.InteropServices" />
    <NetCoreReference Include="System.Security.Cryptography.Algorithms" />
    <NetCoreReference Include="System.Security.Cryptography.Primitives" />
    <NetCoreReference Include="System.Security.Cryptography.X509Certificates" />
    <NetCoreReference Include="System.Threading" />
    <NetCoreReference Include="System.Threading.Thread" />
    <NetCoreReference Include="System.Threading.ThreadPool" />

    <!-- Additional references beyond the minimal set for PresentationFramework.Aero -->
    <NetCoreReference Include="System.ComponentModel.TypeConverter" />
    <NetCoreReference Include="System.Collections.Concurrent" />
  </ItemGroup>
```

Reply:
Maybe I am late on this, but do we have a binlog from a prior build that proves they were getting the wrong WindowsBase in wpf-int? If the answer is no, then I don't think we should make this change, but fix the underlying reason why this has changed in the port to GitHub.

Reply:
Something must've broken here, because it looks like all the references from `Micrososft.NetCore.App.Ref` are being passed to the compiler, not just the ones specified by `NetCoreReference`.

I'd prefer to just keep this change, rather than try to unwind what all of these targets are trying/supposed to do and figure out what exactly is going on. I've spent 20-30 minutes looking through the binlogs and nothing really stands out other than things don't seem to be working the way they should. I'm not even convinced that this is an incorrect fix, it's simple and straightforward.

Reply:
I don't agree, if the targets work 100% on GitHub and on wpf-int and this change breaks them, then this change should fix them directly.

Adding stuff on top of the engineering system without truly knowing the root of the break isn't something I'm willing to sign off on personally.  I'll defer to @vatsan-madhavan on this if he thinks it's correct, but to me this feels like doing more work on top of an issue we already handled historically and doesn't exist outside of this PR.

If I'm characterizing this error wrong, someone please explain to me what I am missing.

Reply:
Let's try to understand what's really going on. I'd like a root-cause for why Microsoft.NetCore.App.Ref is being involved when we expect it to be filtered out.

Reply:
See my comment here https://github.com/dotnet/wpf/pull/1576#discussion_r312260083

Reply:
Not quite.

Please cherry-pick https://github.com/dotnet/wpf/commit/e00c96ad09c93306fcd59214883deec025f742d9 and undo your Pbt.targets changes; then try again.